extensions/kafka/runtime/src/main/java/org/apache/camel/quarkus/component/kafka/KafkaClientFactoryProducer.java

```diff
@@ -33,13 +35,30 @@
     @Identifier("default-kafka-broker")
     Map<String, Object> kafkaConfig;
 
+    @Inject
+    CamelKafkaRuntimeConfig camelKafkaRuntimeConfig;
+
+    @Inject
+    Config config;
+
     @Produces
     @Singleton
     @DefaultBean
-    public KafkaClientFactory kafkaClientFactory(CamelKafkaRuntimeConfig config) {
-        if (kafkaConfig != null && !kafkaConfig.isEmpty() && config.kubernetesServiceBinding.mergeConfiguration) {
+    public KafkaClientFactory kafkaClientFactory() {
+        if (isQuarkusKafkaClientFactoryRequired()) {
             return new QuarkusKafkaClientFactory(kafkaConfig);
         }
         return null;
     }
+
+    private boolean isQuarkusKafkaClientFactoryRequired() {
+        Optional<Boolean> serviceBindingEnabled = config.getOptionalValue(
+                "quarkus.kubernetes-service-binding.enabled",
+                Boolean.class);
```

Comment:
FWIW - Ideally I don't want to be looking up config values in this way. But the current state of `quarkus-kubernetes-service-binding` mandates it. I have a more optimised of doing the `KafkaClientFacotry` setup, but it would need some changes on the Quarkus side.

Reply:
ok, that's fine. Thanks for explanation :+1: